src/Microsoft.Identity.Client/Internal/Requests/BrokerInteractiveRequest.cs

```diff
@@ -0,0 +1,76 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.Utils;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Internal.Requests
+{
+    internal class BrokerInteractiveRequest : RequestBase
+    {
+        BrokerFactory brokerFactory = new BrokerFactory();
+        IBroker BrokerHelper;
+        AcquireTokenByBrokerParameters _brokerParameters;
+
+        public BrokerInteractiveRequest(
+           IServiceBundle serviceBundle,
+           AuthenticationRequestParameters authenticationRequestParameters,
+           AcquireTokenByBrokerParameters brokerParameters)
+           : base(serviceBundle, authenticationRequestParameters, brokerParameters)
+        {
+            BrokerHelper = brokerFactory.CreateBrokerFacade(ServiceBundle.DefaultLogger);
+            _brokerParameters = brokerParameters;
```

Comment:
What does this do ? It looks like you are initializing the broker parameters from other params anyway. Is there a need for  AcquireTokenByBrokerParameters

Reply:
changing this. 

---
In reply to: [254646290](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/874#discussion_r254646290) [](ancestors = 254646290)